### PR TITLE
*: add session var 'tidb_ddl_reorg_worker_cnt' to control ddl reorg workers count

### PR DESCRIPTION
**Pull request by winkyao**
After this PR, you can use below command to modify the count of ddl reorg workers, to 
increase add index concurrency:

```
set tidb_ddl_reorg_worker_cnt = 100
```

### CODE REVIEW

File: executor/set.go [hunk 1]
```diff
@@ -186,6 +186,8 @@ func (e *SetExecutor) setSysVariable(name string, v *expression.VarAssignment) e
 		if isoLevel == ast.ReadCommitted {
 			e.ctx.Txn().SetOption(kv.IsolationLevel, kv.RC)
 		}
+	} else if name == variable.TiDBDDLReorgWorkerCount {
+		domain.GetDomain(e.ctx).DDL().WorkerVars().DDLReorgWorkerCount = sessionVars.DDLReorgWorkerCount
```
Comment (shenli) [inline executor/set.go]: If this is a session-scope variable, we have to find the instance which is ddl-owner and run the set statement on it. We should make it a global scope variable.
(And actually, it is a server-scope variable.)

**Comment by winkyao** [comment]
@shenli Done

**Comment by shenli** [comment]
Please fix the CI.

File: sessionctx/variable/tidb_vars.go [hunk 2]
```diff
@@ -189,9 +192,11 @@ const (
...
 	DefTiDBHashJoinConcurrency       = 5
 	DefTiDBOptimizerSelectivityLevel = 0
+	DefTiDBDDLReorgWorkerCount       = 16
 )
 
 // Process global variables.
 var (
-	ProcessGeneralLog uint32
+	ProcessGeneralLog     uint32
+	DDLReorgWorkerCounter int32 = DefTiDBDDLReorgWorkerCount
```
Comment (zimulala) [inline sessionctx/variable/tidb_vars.go]: s/DDLReorgWorkerCounter/ddlReorgWorkerCounter ?
We use `GetDDLReorgWorkerCounter` and `SetDDLReorgWorkerCounter` to change its value. So I think we'd better use the lowercase variable.

File: sessionctx/variable/session.go [hunk 3]
```diff
@@ -531,6 +531,9 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.EnableStreaming = TiDBOptOn(val)
 	case TiDBOptimizerSelectivityLevel:
 		s.OptimizerSelectivityLevel = tidbOptPositiveInt32(val, DefTiDBOptimizerSelectivityLevel)
+	case TiDBDDLReorgWorkerCount:
+		workerCnt := tidbOptPositiveInt32(val, DefTiDBDDLReorgWorkerCount)
```
Comment (zimulala) [inline sessionctx/variable/session.go]: Do we need to have an upper limit?

**Comment by jackysp** [comment]
Please resolve the conflicts @winkyao .

**Comment by shenli** [comment]
@winkyao Please resolve the conflicts.

File: sessionctx/variable/session.go (hunk 3 above)
Comment (winkyao) [reply sessionctx/variable/session.go]: Done

**Comment by winkyao** [comment]
@zimulala @shenli @jackysp PTAL

File: ddl/index.go [hunk 4]
```diff
@@ -837,10 +837,10 @@ func (d *ddl) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgInfo
 	log.Infof("[ddl-reorg] addTableIndex, job:%s, reorgInfo:%#v", job, reorgInfo)
 	colFieldMap := makeupIndexColFieldMap(t, indexInfo)
 
-	// TODO: make workerCnt can be modified by system variable.
-	workerCnt := defaultWorkers
+	// variable.ddlReorgWorkerCounter can be modified by system variable "tidb_ddl_reorg_worker_cnt".
+	workerCnt := variable.GetDDLReorgWorkerCounter()
```
Comment (zimulala) [inline ddl/index.go]: Could we support updating worker counter when the workers are running?

**Comment by shenli** [comment]
Please resolve the conflicts.

**Comment by winkyao** [comment]
@zimulala @shenli PTAL

**Comment by crazycs520** [comment]
LGTM

**Pull request merged**